packages/terra-dynamic-grid/src/styles.js

```diff
@@ -0,0 +1,110 @@
+// IE < 11 doesn't honor window.CSS.Supports. Edge has supports, but will fail on grid.
+const cssGridsEnabled = window.CSS && window.CSS.supports && window.CSS.supports('display', 'grid');
+
+const gridTemplateColumns = (layout) => {
+  const columns = layout['grid-template-columns'];
+  const msColumns = columns.split(/\s+/).join(` ${layout['grid-gap'] || 0} `);
+
+  return {
+    'grid-template-columns': columns,
+    '-ms-grid-columns': msColumns,
+  };
+};
+
+const gridTemplateRows = (layout) => {
+  const rows = layout['grid-template-rows'];
+  const msRows = rows.split(/\s+/).join(` ${layout['grid-gap'] || 0} `);
+
+  return {
+    'grid-template-rows': rows,
+    '-ms-grid-rows': msRows,
+  };
+};
+
+
+const gridColumnStart = (layout, region) => {
+  const start = region['grid-column-start'];
+  const msStart = (start * 2) - 1;
+
+  return {
+    'grid-column-start': `${start}`,
+    '-ms-grid-column': `${msStart}`,
+  };
+};
+
+const gridColumnEnd = (layout, region) => {
+  const start = region['grid-column-start'];
+  const end = region['grid-column-end'] || start + 1;
+  const span = end - start < 2
+    ? 1 // No gaps traversed
+    : (end - start) * 2;
+
+  return {
+    'grid-column-end': `${end}`,
+    '-ms-grid-column-span': `${span}`,
+  };
+};
+
+
+const gridRowStart = (layout, region) => {
+  const start = region['grid-row-start'];
+  const msStart = (start * 2) - 1;
+
+  return {
+    'grid-row-start': `${start}`,
+    '-ms-grid-row': `${msStart}`,
+  };
+};
+
+const gridRowEnd = (layout, region) => {
+  const start = region['grid-row-start'];
+  const end = region['grid-row-end'] || start + 1;
+  const span = end - start < 2
+    ? 1 // No gaps traversed
+    : (end - start) * 2;
+
+  return {
+    'grid-row-end': `${end}`,
+    '-ms-grid-row-span': `${span}`,
+  };
+};
+
+
+const generateGridStyles = (layout) => {
+  const styles = {
+    display: cssGridsEnabled ? 'grid' : '-ms-grid',
+    'grid-gap': layout['grid-gap'],
+    ...gridTemplateColumns(layout),
+    ...gridTemplateRows(layout),
+  };
+
+  return { grid: layout.media
+    ? { [layout.media]: styles }
+    : styles };
+};
+
+const generateRegionStyles = layout => (
+  (region) => {
+    const styles = {
+      ...gridColumnStart(layout, region),
```

Comment:
I think i can remove layout from being passed in.. I though I needed to have access to it to determine if I needed to calculate grid gaps, but to keep things simple, I just always include grid gaps and set them to 0.